homeassistant/components/workday/config_flow.py

```diff
@@ -0,0 +1,291 @@
+            subdiv=user_input[CONF_PROVINCE], years=year
+        )
+
+    for remove_date in user_input[CONF_REMOVE_HOLIDAYS]:
+        if dt.parse_date(remove_date) is None:
+            if obj_holidays.get_named(remove_date) == []:
+                raise RemoveDatesError("Incorrect date or name")
+
+
+DATA_SCHEMA_SETUP = vol.Schema(
+    {
+        vol.Required(CONF_NAME, default=DEFAULT_NAME): TextSelector(),
+        vol.Required(CONF_COUNTRY): SelectSelector(
+            SelectSelectorConfig(
+                options=list(holidays.list_supported_countries()),
+                mode=SelectSelectorMode.DROPDOWN,
+            )
+        ),
+    }
+)
+
+DATA_SCHEMA_OPT = vol.Schema(
+    {
+        vol.Optional(CONF_EXCLUDES, default=DEFAULT_EXCLUDES): SelectSelector(
+            SelectSelectorConfig(
+                options=ALLOWED_DAYS, multiple=True, mode=SelectSelectorMode.DROPDOWN
+            )
+        ),
+        vol.Optional(CONF_OFFSET, default=DEFAULT_OFFSET): NumberSelector(
+            NumberSelectorConfig(min=-10, max=10, step=1, mode=NumberSelectorMode.BOX)
+        ),
+        vol.Optional(CONF_WORKDAYS, default=DEFAULT_WORKDAYS): SelectSelector(
+            SelectSelectorConfig(
+                options=ALLOWED_DAYS, multiple=True, mode=SelectSelectorMode.DROPDOWN
+            )
+        ),
+        vol.Optional(CONF_ADD_HOLIDAYS, default=[]): SelectSelector(
+            SelectSelectorConfig(
+                options=[],
+                multiple=True,
+                custom_value=True,
+                mode=SelectSelectorMode.DROPDOWN,
+            )
+        ),
+        vol.Optional(CONF_REMOVE_HOLIDAYS, default=[]): SelectSelector(
+            SelectSelectorConfig(
+                options=[],
+                multiple=True,
+                custom_value=True,
+                mode=SelectSelectorMode.DROPDOWN,
+            )
+        ),
+    }
+)
+
+
+class WorkdayConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Workday integration."""
+
+    VERSION = 1
+
+    data: dict[str, Any] = {}
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: ConfigEntry,
+    ) -> WorkdayOptionsFlowHandler:
+        """Get the options flow for this handler."""
+        return WorkdayOptionsFlowHandler(config_entry)
+
+    async def async_step_import(self, config: dict[str, Any]) -> FlowResult:
+        """Import a configuration from config.yaml."""
+
+        input_config = {
+            k: v for k, v in config.items() if k not in [CONF_NAME, CONF_PLATFORM]
+        }
+        self._async_abort_entries_match(input_config)
+        return await self.async_step_options(user_input=config)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the user initial step."""
+        if user_input is not None:
+            self.data = user_input
+            return await self.async_step_options()
```

Comment:
Should you not validate the country here, making sure it exists in the library?

Reply:
Tests needs to be added before I put this for review but appreciate if you review the code before I add the tests in a while

Reply:
Already taken care off by the schema so not needed